### ISSUE TITLE
CosmosClient init - AttributeError: 'NoneType' object has no attribute 'ConsistencyPolicy'

### ISSUE THREAD

**Issue by aj9411** [Azure/azure-sdk-for-python#37658]
- **Package Name**: azure.cosmos
- **Package Version**: 4.3.1
- **Operating System**: Linux Mariner
- **Python Version**: 3.8

**Describe the bug**
Creating a CosmosClient with `CosmosClient(endpoint, key)` I am getting this error `AttributeError: 'NoneType' object has no attribute 'ConsistencyPolicy'`

a bit more from the stack trace:

```
client = cosmos_client.CosmosClient(CosmosDbURI, {"masterKey": CosmosDbPrimaryKey}) (this first line is my code)
File "/usr/local/lib/python3.8/dist-packages/azure/cosmos/cosmos_client.py", line 184, in __init__
   self.client_connection = CosmosClientConnection(
File "/usr/local/lib/python3.8/dist-packages/azure/cosmos/_cosmos_client_connection.py", line 214, in __init__
   self._set_client_consistency_level(database_account, consistency_level)
File "/usr/local/lib/python3.8/dist-packages/azure/cosmos/_cosmos_client_connection.py", line 231, in _set_client_consistency_level
   user_consistency_policy = database_account.ConsistencyPolicy
AttributeError: 'NoneType' object has no attribute 'ConsistencyPolicy'
```

I see on this file https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py, ` def _set_client_consistency_level` this `if` block:

```
        if consistency_level is None and database_account.ConsistencyPolicy:
            # Set to default level present in account
            user_consistency_policy = database_account.ConsistencyPolicy
            consistency_level = user_consistency_policy.get(Constants.DefaultConsistencyLevel)
```

I think the exception is from here... It doesn't make sense why we are getting into the body of the `if` if the `database_account` is `None`? The CosmosDB account I am trying to connect and has default consistency of session (see screenshot)

**To Reproduce**
1. Deploy CosmosDB account, set it to default consistency to session
2. Try to create a CosmosClient with the endpoint and key, no consistency level

**Expected behavior**
I can initialize the CosmosClient

**Screenshots**
CosmosDB account exists and has session as the default consistency
![image](https://github.com/user-attachments/assets/3d8aa095-9656-40de-ba02-8619677b6b30)

**Additional context**
Related post https://github.com/Azure/azure-sdk-for-python/issues/27547

**Comment by kashifkhan** [comment]
Thank you for the feedback @aj9411 . We will investigate and get back to you asap.

**Comment by aj9411** [comment]
I'll add a couple more things, I changed my code to explicitly add the consistency level, and then I got a new error saying I couldn't use keys to authenticate. I noticed that there was a change in the CosmosDB account, it switched to have local auth (key-access) disabled as we are in the process of moving away from keys. I re-enabled key access, and I was able to run fine. Maybe the initial error happens when local auth is disabled and there's no consistency level specified? If that's the case, maybe a better error would help.
